packages/server/src/StoredFaucetService.ts

```diff
@@ -88,16 +98,21 @@ export class StoredFaucetService {
    */
   private async checkState() {
     try {
-      const LFB = await this.casperService.getLastFinalizedBlockInfo();
-      const blockHash = LFB.getSummary()!.getBlockHash_asU8();
-      const stateQuery = new StateQuery();
-      stateQuery.setKeyBase16(encodeBase16(this.contractKeys.publicKey));
-      stateQuery.setKeyVariant(StateQuery.KeyVariant.ADDRESS);
-      stateQuery.setPathSegmentsList([CONTRACT_NAME]);
+      const LFB = await this.casperService.getLatestBlockInfo().then(it => {
+        return it.block;
+      });
+      if (!LFB) {
+        return false;
+      }
+      const globalStateHash = LFB.header!.state_root_hash!;
 
+      const accountHash = Keys.Ed25519.publicKeyHash(
+        this.contractKeys.publicKey
+      );
       const state = await this.casperService.getBlockState(
-        blockHash,
-        stateQuery
+        globalStateHash,
+        'account-hash-' + encodeBase16(accountHash),
```

Comment:
I see this for the second time. Make the `'account-hash-' + something` a function.